app.go

```diff
@@ -839,12 +840,30 @@ func (app *App) HandlersCount() uint32 {
 }
 
 // Shutdown gracefully shuts down the server without interrupting any active connections.
-// Shutdown works by first closing all open listeners and then waiting indefinitely for all connections to return to idle and then shut down.
+// Shutdown works by first closing all open listeners and then waiting indefinitely for all connections to return to idle before shutting down.
 //
 // Make sure the program doesn't exit and waits instead for Shutdown to return.
 //
 // Shutdown does not close keepalive connections so its recommended to set ReadTimeout to something else than 0.
 func (app *App) Shutdown() error {
+	return app.shutdownWithContext(context.Background())
+}
+
+// ShutdownWithTimeout gracefully shuts down the server without interrupting any active connections. However, if the timeout is exceeded,
+// ShutdownWithTimeout will forcefully close any active connections.
+// ShutdownWithTimeout works by first closing all open listeners and then waiting for all connections to return to idle before shutting down.
+//
+// Make sure the program doesn't exit and waits instead for ShutdownWithTimeout to return.
+//
+// ShutdownWithTimeout does not close keepalive connections so its recommended to set ReadTimeout to something else than 0.
+func (app *App) ShutdownWithTimeout(timeout time.Duration) error {
```

Comment:
I think we can make ShutdownWithContext public instead. Or add variadic context.Context param to Shutdown method. What do you think @ReneWerner87 @li-jin-gou

Reply:
I think so.

Reply:
I think allowing context is overkill. Context allows for cancellation, deadline or timeout. This function only makes sense with a timeout.

Reply:
I think we may need to pass context for some cases

Reply:
> I think we may need to pass context for some cases

Can you produce some realworld example scenarios so I can understand the background better?

functionality seems to me actually okay, don't know if you really still need the context or maybe you have already prepared the context outside and want to use it then, but don't know if this is a real world scenario or just a very theoretical one

Reply:
I think it would be really useful to use https://pkg.go.dev/os/signal#NotifyContext easily

Reply:
I think there is a misunderstanding of what this function does. `https://pkg.go.dev/os/signal#NotifyContext` can/should be used to _trigger_ the `Shutdown` function. It has nothing to do with the timeout for when the force shutdown should be activated.